### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by kazumaki**
We had to create another repository to fix stickler, it was not working on our last one.

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -1,218 +1,133 @@
...
+      system('clear')
+      puts 'Displaying Board...'
+      game.sleep_mode
+
+      game.board.reset
+      puts game.draw_board
+      game.game_status(true)
+    end
+  end
+  current_player = current_player.name == game.board.player1.name ? game.board.player2 : game.board.player1
```
Comment (maelfosso) [inline bin/main.rb]: You could put it into the `Game` class
1. Create a property  (reader) **current_player**
2. Create a method **switch_player** for example
3. And
```ruby
def switch_player
    @current_player = @current_player.name == player1.name ? player2 : player1
end
```

File: bin/main.rb [hunk 2]
```diff
@@ -1,218 +1,133 @@
...
+marker1 = gets.chomp.upcase.to_sym
+until [:X, :O].include? marker1
+  game.sleep_mode
+  puts "Invalid Entry #{player_first.capitalize}, please re-enter your marker: "
+  marker1 = gets.chomp.upcase.to_sym
+end
+
+marker2 = marker1 == :X ? :O : :X
+
+game.start_board(player_first, player_second, marker1, marker2)
```
Comment (maelfosso) [inline bin/main.rb:64]: Constructor of **Game** class

File: bin/main.rb [hunk 3]
```diff
@@ -1,218 +1,133 @@
...
+
+current_player = player_first == game.board.player1.name ? game.board.player1 : game.board.player2
```
Comment (maelfosso) [inline bin/main.rb]: move it into the constructor of **Game** class

File: lib/board.rb [hunk 4]
```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require_relative('player.rb')
+
+class Board
+  attr_reader :cell_grid, :available, :player1, :player2
+  def initialize(player1, player2, marker1, marker2)
+    reset
+    @player1 = Player.new(player1, marker1)
+    @player2 = Player.new(player2, marker2)
```
Comment (maelfosso) [inline lib/board.rb:10]: Move it into the **Game** class
The Players are part of the Game not of the Board

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required

Hello coders,

Really nice code, nice game boards, nice nice nice :clap: 
However just a few things to changes.

- Make **bin/main.rb** executable such that we could run the game by only entering `bin/main.rb`

- There is no message when the game is a **Tie / Draw**
```ruby
Available      TIC-TAC_TOE
moves

  |   |         X | X | O
---------       --------- 
  |   |         O | O | X
---------       --------- 
  |   |         X | O | X


Do you want to play again (y/n)
```
- Read the comment below to fix some changes

Kindly make the changes and submit a new code review request.
Thanks

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

**Review by whiz25 — APPROVED** [review]
## Cool! :clap: 
I tested your app, it works properly. All requirements met so it's a :+1:  from me.

### Martin Afani

**Pull request merged**